### ISSUE TITLE
Won't retry, spits out Failed! (Network Error)

### ISSUE THREAD

**Issue by pencilbook** [ccloli/E-Hentai-Downloader#160]
It'll say "pending" and won't even attempt to retry and then it'll return "Failed! (Network Error)"

**Comment by ccloli** [comment]
Nope, in fact it's retrying but with no progress (it will keep the previous status until it downloads some content).      So when you see `Pending...` means it's now start downloading but doesn't receive anything yet, and yellow `Failed (...)` means it's now retrying to download again but with no progress too. Only red `Failed (...)` means it exceeds the maximum retry times.      label:question     ------------------ Original ------------------

**Comment by pencilbook** [comment]
Oh I see. 

I'm pretty sure it used to say "retrying" before, was there a reason for it to change?

**Comment by ccloli** [comment]
Well, `Pending...` means it's _downloading_ but **haven't received data** yet, and `Downloading...` means it **receives some data** now.

So just like the normal status, `Failed` status that shows as yellow means it _stops old request_ and _starts a new request_ but **no data**, and `Retrying...` means the request **has got some data**.

```
               get
               some
 +---------+   data  +-------------+  finish  +---------+
 | Pending +----+----> Downloading +----------> Succeed <-------+
 +---------+    |    +------+------+          +---------+       |
loading url &   |           |        get error                  |
start request   | get error +-----------------------+           |
                |           |                       |           |
                |       +---v----+            +-----+----+      |
                +-------> Failed +------------> Retrying +------+
                        +--------+     get    +----------+ finish
                     abort request &   some
                     start retrying    data

```

**Comment by pencilbook** [comment]
I am not sure if there's something wrong with my settings. Image 4, 14, 20, and 35 aren't loading at all for me. When I try to read those pages from the site, it won't load until I click on "Click here if the image fails loading" under the image. I always assumed when the script would retry, it would download the pages from the "Click here if the image fails loading" links.

Gallery: e-hentai.org/g/1620312/9cf711ab6f/

**Comment by pencilbook** [comment]
Suggestion: When the script fails to download an image after one attempt, it should download the image from "Click here if the image fails loading"

**Comment by ccloli** [comment]
> Suggestion: When the script fails to download an image after one attempt, it should download the image from "Click here if the image fails loading"

Yes it will, once it hits the maximum retry time (default is `3`) and you choose retry downloading. That means before the `Failed` goes red, it won't try that link as it may cost you more viewing limits (it hits original image server instead of H@H nodes). When other images are done, script will ask you to retry failed ones, that'll use the `Click here if the image fails loading` link.

**Comment by pencilbook** [comment]
I don't think that's happening anymore, a lot has changed on the site. I can retrieve the failed image by clicking on "Click here if the image fails loading", so there should be no reason for the script to fail. What's your FF version and are you using Greasemonkey or Violentmonkey?

**Comment by ccloli** [comment]
Maybe I didn't explain it well, so let's make it short.

When it download fails, it'll attempt to retry 3 times by default, at this time it **won't** reach `Click here if the image fails loading` link. If all of these retrys are failed, the script will ask you to try again, at this time it **will** reach `Click here if the image fails loading` link<sup>*</sup>.

*: unless you enabled _Never get new image URL when failed to download image_ or _Never send "nl" GET parameter when getting new image URL_, which are disabled by default.
